jupyter_notebooks/Book_4_Mountain_Wave_Moving_Mesh.ipynb

```diff
@@ -0,0 +1,508 @@
+   "metadata": {},
+   "source": [
+    "This case will differ from the sk_nonlinear case we saw in Notebook 2, as a hydrostatic balance is no longer enforced."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "29df76b1",
+   "metadata": {},
+   "source": [
+    "We begin by importing the required libraries and functions from Firedrake and Gusto:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "7779c15b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "firedrake:WARNING OMP_NUM_THREADS is not set or is set to a value greater than 1, we suggest setting OMP_NUM_THREADS=1 to improve performance\n"
+     ]
+    }
+   ],
+   "source": [
+    "from gusto import *\n",
+    "from firedrake import (as_vector, VectorFunctionSpace,\n",
+    "                       PeriodicIntervalMesh, ExtrudedMesh, SpatialCoordinate,\n",
+    "                       exp, pi, cos, Function, conditional, Mesh, op2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "33f71f3a",
+   "metadata": {},
+   "source": [
+    "Define the time-step size and the duration of the simulation. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "f5147739",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dt = 5.0\n",
+    "\n",
+    "tmax = 9000.\n",
+    "dumpfreq = int(tmax / (9*dt))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "48ee6e88",
+   "metadata": {},
+   "source": [
+    "Like with the gravity wave example, we will create a periodic interval mesh in the horizontal and extrude this in the vertical dimension to create a three-dimensional mesh."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "844be5eb",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "UFL:WARNING Discontinuous Lagrange element requested on interval * interval, creating DQ element.\n"
+     ]
+    }
+   ],
+   "source": [
+    "nlayers = 70  # horizontal layers\n",
+    "columns = 180  # number of columns\n",
+    "L = 144000.\n",
+    "m = PeriodicIntervalMesh(columns, L)\n",
+    "\n",
+    "# build volume mesh\n",
+    "H = 35000.  # Height position of the model top\n",
+    "ext_mesh = ExtrudedMesh(m, layers=nlayers, layer_height=H/nlayers)\n",
+    "Vc = VectorFunctionSpace(ext_mesh, \"DG\", 2)\n",
+    "coord = SpatialCoordinate(ext_mesh)\n",
+    "x = Function(Vc).interpolate(as_vector([coord[0], coord[1]]))\n",
+    "a = 1000.\n",
+    "xc = L/2.\n",
+    "x, z = SpatialCoordinate(ext_mesh)\n",
+    "hm = 1.\n",
+    "zs = hm*a**2/((x-xc)**2 + a**2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2fb1fb2d",
+   "metadata": {},
+   "source": [
+    "We now invoke the smooth_z variable, I assume to smooth out the vertical profile somehow (do we want an explanation of this?? We then apply this smoothing to the defined mesh."
```

Comment:
I actually think it's clearest if we remove the `smoother_z` flag? But we can still use the smoother z profile. The description here could then be something like: "Now we transform the mesh to follow the defined terrain. A smoothing is applied in the vertical direction."